examples/datasets/plot_iris_dataset.py

```diff
@@ -19,37 +19,50 @@
 # Modified for documentation by Jaques Grobler
 # License: BSD 3 clause
 
-import matplotlib.pyplot as plt
-
-# unused but required import for doing 3d projections with matplotlib < 3.2
-import mpl_toolkits.mplot3d  # noqa: F401
-
+# %%
+# Loading the iris dataset
+# ------------------------
 from sklearn import datasets
-from sklearn.decomposition import PCA
 
-# import some data to play with
 iris = datasets.load_iris()
-X = iris.data[:, :2]  # we only take the first two features.
-y = iris.target
 
-x_min, x_max = X[:, 0].min() - 0.5, X[:, 0].max() + 0.5
-y_min, y_max = X[:, 1].min() - 0.5, X[:, 1].max() + 0.5
 
+# %%
+# Scatter Plot of the Iris dataset
+# --------------------------------
+import matplotlib.pyplot as plt  # noqa: E402
```

Comment:
```suggestion
import matplotlib.pyplot as plt
```

Reply:
I had to leave this in, otherwise the linter would not let me commit

Reply:
This is weird, because our global conf already discard E402.
I committed and pushed with the fix and it seems that pre-commit did not annoy me. Not sure why you got in trouble.